pkg/signals/signal_test.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2020, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package signals
+
+import (
+	"github.com/stretchr/testify/assert"
+	"os"
+	"syscall"
+	"testing"
+)
+
+// TestSetupSignalHandler tests that the Signal Handler returned is valid
+// GIVEN that a valid Signal Handler Channel is returned
+// WHEN I send a SIGTERM to the Process
+// THEN a value should be received from the SignalHandler Channel
+func TestSetupSignalHandler(t *testing.T) {
+	tests := []struct {
+		name string
+	}{
+		{
+			name: "verifySetupSignalHandler",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert := assert.New(t)
+
+			gotStopCh := SetupSignalHandler()
+			assert.NotNil(gotStopCh, "expect the returned signal handler to be not nil")
+			assert.IsType(make(<-chan struct{}), gotStopCh, "expect the returned signal handler to be of type <-chan struct{}")
```

Comment:
Is it possible for the returned value to be any other type?
```
func SetupSignalHandler() (stopCh <-chan struct{}) {
```
Is this really checking anything useful?

Reply:
what it is checking, and I may be overly cautious here, is that someone will update the function to return more that an empty struct
There is this in controller.go
`	// Set up signals so we handle the first shutdown signal gracefully
	glog.V(6).Info("Setting up signals")
	stopCh := signals.SetupSignalHandler()

	go kubeInformerFactory.Start(stopCh)
	go verrazzanoOperatorInformerFactory.Start(stopCh)
	go wlsoprInformerFactory.Start(stopCh)
	go vmoInformerFactory.Start(stopCh)`

And the signal channel looks to be passed all over the place and w go's loose type checking, thought I would catch it early